src/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -341,6 +341,128 @@ public BitArray Not()
             return this;
         }
 
+        /*=========================================================================
+        ** Shift all the bit values to right on count bits. When count less zero it
+        ** shift all the bit values to left. New bits are be set to zero. The 
+        ** current instance is updated and returned.
+        =========================================================================*/
+        public BitArray RightShift(int count)
+        {
+            if (count <= 0)
+            {
+                if (count == 0)
+                {
+                    _version++;
+                    return this;
+                }
+                count = -count;
+                // Following trick reduce check with int.MinValue
+                if (unchecked((uint)count < (uint)m_length))
```

Comment:
Why all the explicit usage of _unchecked_ when this is already the default for non-constants?